### PR TITLE
fix(scipy.special): zeta returns NaN for q=inf; return 0.0 instead (#…

### LINKED ISSUE #38637: jax.lax.zeta returns NaN for q=inf where zero is expected
### Description

`jax.lax.zeta` appears to return `NaN` when `q` is positive infinity.

For the Hurwitz zeta function, when `s > 1` and `q = inf`, the result should tend to `0.0`. However, JAX returns `NaN` for this valid limiting case.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import warnings

warnings.filterwarnings("ignore")
os.environ.setdefault("JAX_PLATFORMS", "cpu")

try:
    import numpy as np
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

s = np.array([2.0], dtype=np.float32)
q = np.array([np.inf], dtype=np.float32)

expected = np.array([0.0], dtype=np.float32)
out = np.asarray(jax.lax.zeta(jnp.asarray(s), jnp.asarray(q)))

print("jax:", out.tolist())
print("expected:", expected.tolist())

if not np.array_equal(out, expected, equal_nan=True):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: [nan]
expected: [0.0]
```

### Expected result

```
[0.0]
```

For `s = 2.0` and `q = inf`, the Hurwitz zeta value should be zero in the limit, not `NaN`.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

### PR DESCRIPTION
**Pull request by Adityakk9031**
Summary

Fixes #38637.

jax.scipy.special.zeta(s, q) returned NaN when q = +inf, even though the correct mathematical limit is 0.0 for all s > 1.

python
# Before this fix
jax.lax.zeta(jnp.float32(2.0), jnp.float32(jnp.inf))
# → nan  
# After this fix
# → 0.0  
Root Cause

The XLA kernel (xla/hlo/builder/lib/math.cc) implements the Hurwitz zeta function via an Euler-Maclaurin expansion. When q = +inf, intermediate temporaries compute neg_power * acc = 0 * inf = NaN due to IEEE 754 arithmetic — XLA has no guard for q = +inf. The bug lives in the C++ backend and is not patchable without rebuilding jaxlib, so we apply a Python-level guard after the XLA call, consistent with how similar edge-cases (igammac for x=inf) are handled elsewhere in JAX's scipy layer.

Fix

Two one-liner jnp.where guards in jax/_src/scipy/special.py:

Forward pass (zeta): return jnp.where(jnp.isposinf(q), jnp.zeros_like(result), result)
JVP helper (_zeta_series_expansion): same guard, so jax.grad(zeta, argnums=1)(s, inf) also returns 0.0 instead of NaN.
The guard composes cleanly with jit/grad/vmap, adds no measurable overhead for finite q, and does not suppress valid NaN outputs (e.g. q = -inf correctly remains NaN).

Changes

File	Change
jax/_src/scipy/special.py	Add q=+inf → 0.0 guard to zeta and _zeta_series_expansion
tests/lax_scipy_special_functions_test.py	Add testZetaQInfinity regression test
Tests — testZetaQInfinity covers: q=+inf → 0.0 (float32 & float64), q=-inf still NaN, batched mixed inputs, under JIT, and gradient at q=inf is 0.0.

References — Closes #38637 · Related: #38199 · Johansson (2015) https://arxiv.org/abs/1309.2877

### CODE REVIEW

**Comment by Adityakk9031** [comment]
@bchetioui have look a please

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
